firewall-disable-with-options.ks.in

```diff
@@ -0,0 +1,57 @@
+#version=DEVEL
+#test name: firewall-disable-with-options
+#
+# Test that firewall can be properly disabled & all the
+# firewall options are still set correctly on the target system.
+%ksappend repos/default.ks
+
+%ksappend common/common_no_payload.ks
+%ksappend payload/default_packages.ks
+
+# TEST: firewall
+firewall --disable --port=22001:tcp,6400:udp --service=tftp,smtp
+
+%post
+# The firewall --disable kickstart command gets translated into firewall-offline-cmd --disable,
+# which simply disables the firewalld systemd unit. So by checking if the unit is disabled,
+# we can check if the kickstart command works correctly.
+systemctl is-enabled firewalld
+if [[ $? -eq 0 ]]; then
+    echo "*** firewalld.service should be disabled" >> /root/RESULT
+fi
+
+# Even though we disable the firewall, we still forward the options
+# to firewall-offline-cmd & they should be set properly.
+
+# Test for 22001/TCP
+firewall-offline-cmd --list-ports | grep 22001/tcp
+if [[ $? -ne 0 ]]; then
+    echo "*** Firewall config for 22001/tcp" >> /root/RESULT
+fi
+
+# Test for 6400/UDP
+firewall-offline-cmd --list-ports | grep 6400/udp
+if [[ $? -ne 0 ]]; then
+    echo "*** Firewall config for 6400/udp failed" >> /root/RESULT
+fi
+
+# Test for service tftp
+firewall-offline-cmd --list-services | grep tftp
+if [[ $? -ne 0 ]]; then
+    echo "*** Firewall service tftp not assigned" >> /root/RESULT
```

Comment:
I would rather wrote ``Firewall has not opened ports for service tftp``.
The same works for smtp and sane services below.

Reply:
Yep, that sounds weird. Reformulated. :)